core/src/test/java/org/opensearch/sql/expression/function/BuiltinFunctionRepositoryTest.java

```diff
@@ -98,6 +128,60 @@ void resolve() {
     assertEquals(functionExpressionBuilder, repo.resolve(functionSignature));
   }
 
+  @Test
+  void resolve_should_not_cast_arguments_in_cast_function() {
+    when(mockExpression.toString()).thenReturn("string");
+    FunctionImplementation function =
+        repo.resolve(registerFunctionResolver(CAST_TO_BOOLEAN.getName(), DATETIME, BOOLEAN))
```

Comment:
Does the test case works for string and boolean also?

Reply:
Yes, it's covered in `resolve_should_cast_arguments` below.